kernel/byterun/coq_interp.c

```diff
@@ -420,7 +469,28 @@ value coq_interprete
 	coq_extra_args = 2;
 	goto check_stack;
       }
-      /* Stack checks */
+
+       Instruct(APPLY4) {
```

Comment:
Why do you need this new instruction?

Reply:
This is an optimization that has proven useful in some computation-intensive settings. I can try to move it to a separate commit if you like.

Reply:
As you wish, I was just wondering.